src/lib.rs

```diff
@@ -338,6 +339,33 @@ impl From<Vec<Vec<i8>>> for ExecutionData {
     }
 }
 
+impl From<Vec<Vec<i16>>> for ExecutionData {
+    fn from(data: Vec<Vec<i16>>) -> Self {
+        // Shots was passed into QVM originally as a u16 so this is safe.
+        #[allow(clippy::cast_possible_truncation)]
+        let number_of_shots = data.len() as u16;
+
+        // This one is a guess. If more than 2^16 slots in a register then this will truncate
+        #[allow(clippy::cast_possible_truncation)]
+        let shot_length = data[0].len() as u16;
+
+        let results: Vec<*mut i16> = IntoIterator::into_iter(data)
+            .map(|mut shot| {
+                shot.shrink_to_fit();
+                ManuallyDrop::new(shot).as_mut_ptr()
+            })
+            .collect();
+
+        // TODO: consider different `DataType` variant here, rather than re-use of smaller `Byte`
```

Comment:
Mind capturing this as an issue instead?

Reply:
https://github.com/rigetti/qcs-sdk-c/issues/12 for posterity